src/hooks/wallets/useOnboard.ts

```diff
@@ -79,16 +79,38 @@ const getWalletConnectLabel = async ({ label, provider }: ConnectedWallet): Prom
   return peerWallet ?? UNKNOWN_PEER
 }
 
-const trackWalletType = async (wallet: ConnectedWallet) => {
-  trackEvent({ ...WALLET_EVENTS.CONNECT, label: wallet.label })
+const prevTrackedWallet = {
+  label: '',
+  wcLabel: '',
+}
+
+const trackWalletType = async (wallet: ReturnType<typeof getConnectedWallet>) => {
+  if (!wallet) {
+    prevTrackedWallet.label = ''
+    prevTrackedWallet.wcLabel = ''
+    return
+  }
+
+  if (wallet.label !== prevTrackedWallet.label) {
```

Comment:
In what cases does it trigger a connection event w/o this check? Not just once per actual connection?

Reply:
The subscription fires again when ENS domains resolve, etc.

![image](https://user-images.githubusercontent.com/20442784/190378732-33921e5d-099c-4484-acdf-1bf82b8d03cb.png)